internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2616,6 +2616,9 @@ func resourceKubernetesClusterRead(d *pluginsdk.ResourceData, meta interface{})
 				d.Set("node_os_channel_upgrade", nodeOSUpgradeChannel)
 			}
 
+			if props.SecurityProfile == nil {
+				props.SecurityProfile = &managedclusters.ManagedClusterSecurityProfile{}
+			}
 			customCaTrustCertList := flattenCustomCaTrustCerts(props.SecurityProfile.CustomCATrustCertificates)
```

Comment:
We nil check `SecurityProfile` at multiple points in the Read where we set security property values into state. 

For consistency could we actually update `flattenCustomCaTrustCerts` to do the nil check instead of doing a general one here 
```suggestion
			customCaTrustCertList := flattenCustomCaTrustCerts(props.SecurityProfile)
```

`flattenCustomCaTrustCerts` would then become
```
func flattenCustomCaTrustCerts(input *managedclusters.ManagedClusterSecurityProfile) []interface{} {
	if input == nil || input.CustomCATrustCertificates == nil {
		return make([]interface{}, 0)
	}

	customCaTrustCertInterface := make([]interface{}, len(*input.CustomCATrustCertificates))

	for index, value := range *input.CustomCATrustCertificates {
		customCaTrustCertInterface[index] = value
	}

	return customCaTrustCertInterface
}
```